crate/multiworld/src/localisation.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    serde::{
+        Deserialize,
+        Serialize,
+    }, std::fmt
+};
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, PartialEq, clap::ValueEnum)]
+#[clap(rename_all = "lower")]
+pub enum Locale {
+    #[default]
+    EN,
+    FR,
+}
+
+impl Locale {
+    pub const ALL: [Locale; 2] = [
+        Locale::EN,
+        Locale::FR,
+    ];
+
+    pub fn message(&self, message: Message) -> &str {
```

Comment:
To support formatted strings, a good tradeoff between convenience and performance is the [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) type:

```suggestion
    pub fn message(&self, message: Message) -> Cow<'static, str> {
```

You can then use `Cow::Borrowed("...")` for string literals and `Cow::Owned(format!("..."))` for formatted strings.